File: apex-examples/hello-world/src/main/resources/helloworld.js

console.log("cwd:" + new java.io.File(".").getAbsolutePath());

var Router = require("vertx-apex-js/router");

var router = Router.router(vertx);

router.route().handler(function(routingContext) {
  routingContext.response().putHeader("content-type", "text/plain").end("Hello World from JS!");
});

var server = vertx.createHttpServer();

server.requestHandler(router.accept).listen(8080);

File: apex-examples/src/main/java/io/vertx/example/apex/HelloWorld.java
package io.vertx.example.apex;

import io.vertx.core.AbstractVerticle;
import io.vertx.ext.apex.Router;

/*
 * @author <a href="http://tfox.org">Tim Fox</a>
 */
public class HelloWorld extends AbstractVerticle {

  @Override
  public void start() throws Exception {

    Router router = Router.router(vertx);

    router.route().handler(routingContext -> {
      routingContext.response().end("Hello World!");
    });

    vertx.createHttpServer().requestHandler(router::accept).listen(8080);
  }
}
